### PR TITLE
Increase float precision for TraceViewer

### LINKED ISSUE #28: Poor plotting precision at long times
Traces plotted by the TraceViewer tend to get blocky for parts of the signal that occur at late times.

For example, in this 90-minute recording (sampled at 5000 Hz), the signals are plotted nice and smooth early in the record:

early-time

By 30 or 40 minutes, spikes start to become noticeably blocky:

middle-time

This gets worse the later we go (here 72 min):

late-time

The traces get smooth again if you jump back to the start of the recording, so it's not dependent on how long ephyviewer has been running.

If I slowly drag the right-mouse button to adjust the temporal zoom, the blocky edges of the plot jitter around:

zoom

This jittering seems to affect the x-coordinates of the points, but not the y-coordinates.

I haven't gone deep into debugging this yet, so I'm not sure what's wrong. I don't know if this is a problem isolated to ephyviewer, or if pyqtgraph has the same issue more generally.

My gut tells me this has something to do with x-coordinates of the points being calculated by the cumulative summation of the sampling period over the huge number of samples that have elapsed late in the record (in my example, by the 40 minute mark there have been 12 million samples). This could lead to accumulated floating point arithmetic errors. I'm not sure why zooming would matter though.

### PR DESCRIPTION
**Pull request by jpgill86**
Fixes #28

### CODE REVIEW

**Comment by jpgill86** [comment]
@samuelgarcia, do you have any concerns about performance regarding this increase in floating point precision, or can I merge?

**Comment by samuelgarcia** [comment]
Wait.
I have to check it.
I forget to read this.
I think we must go for float64 for time vect but get float32 for data.
I need to make some becnhamrk.

**Comment by jpgill86** [comment]
OK, that sounds reasonable.

File: ephyviewer/traceviewer.py [hunk 1]
```diff
@@ -273,7 +273,7 @@ def get_data(self, t, t_start, t_stop, gains, offsets, visibles, decimation_meth
         
         #~ print(ds_ratio)
         t_start2 = self.source.index_to_time(i_start)
-        times_curves = np.arange(data_curves.shape[1], dtype='float32')
+        times_curves = np.arange(data_curves.shape[1], dtype='float')
```
Comment (samuelgarcia) [inline ephyviewer/traceviewer.py]: Could you put explicitly float64 and add a comment to explain why ?

File: ephyviewer/traceviewer.py [hunk 2]
```diff
@@ -236,8 +236,8 @@ def get_data(self, t, t_start, t_stop, gains, offsets, visibles, decimation_meth
         
         #~ print('sigs_chunk.shape', sigs_chunk.shape)
         data_curves = sigs_chunk[:, visibles].T.copy()
-        if data_curves.dtype!='float32':
-            data_curves = data_curves.astype('float32')
+        if data_curves.dtype!='float':
+            data_curves = data_curves.astype('float')
```
Comment (samuelgarcia) [inline ephyviewer/traceviewer.py]: Let revert that change for now.
I need for inverstigation.

**Comment by samuelgarcia** [comment]
@jpgill86 
Thank you for taking care of this.
I already see this bug in tridesclous, I thought it was fixed here too.

As I said, I propose to force float64 for time only for now.

**Comment by jpgill86** [comment]
How's that?

**Comment by jpgill86** [comment]
I'm going to merge. If I misunderstood what you wanted, we can undo. 😁

**Pull request merged**